lib/portage/manifest.py

```diff
@@ -158,17 +157,16 @@ def __init__(
                 stacklevel=2,
             )
 
-        if find_invalid_path_char is None:
+        if not find_invalid_path_char:
             find_invalid_path_char = _find_invalid_path_char
         self._find_invalid_path_char = find_invalid_path_char
         self.pkgdir = _unicode_decode(pkgdir).rstrip(os.sep) + os.sep
-        self.fhashdict = {}
         self.hashes = set()
         self.required_hashes = set()
 
-        if hashes is None:
+        if not hashes:
```

Comment:
Just for my own Python knowledge—these are not completely equivalent, right?

````
>>> a = ""
>>> a is None
False
>>> not a
True
````

Reply:
That's correct. The first boolean expression tries to identify if the str object `""`, that `a` holds, is the same object in memory as the `None` object. Since `None` is a global static object and of `NoneType`, the only way to get a `True` result is `None is None`. When I'm refactoring, this is what's on my mind:
```python
>>> a = ""
>>> not a
True
>>> a = None
>>> not a
True
>>> a is None
True
```

Reply:
Since `is None` is more selective and evidently the code works fine, I don't think we should change these (unless there's something I'm missing).

Reply:
That's a good point. Rather than me thinking in `bool(Optional[str])` and `bool(str)`, the original code must have explicitly wanted only the former.

Reply:
(It's okay to change stuff ilke this, but do it in a new commit if we want to do it - which is a separate discussion almost - so it's easier to bisect/revert/notice it. I don't have any comment beyond what you guys have already said here though ))